### PR TITLE
[Agent] Add plugin for ActiveMQ 5.x

### PR DESCRIPTION
**Pull request by WithLin**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [ ] Bug fix
- [x] New feature provided
- [ ] Improve performance

- Related issues

___
### Bug fix
- Bug description.

- How to fix?

___
### New feature or improvement
- Describe the details and related test reports.

### CODE REVIEW

**Comment by wu-sheng** [comment]
@WithLin CI is still broken.

@ascrutae Please run this pr with test case pr: https://github.com/SkywalkingTest/skywalking-agent-testcases/pull/20

**Comment by WithLin** [comment]
hi, I have solved the project file, but CI still reported the same error.

File: apm-sniffer/apm-sdk-plugin/activemq-5.x-plugin/src/main/java/org.apache.skywalking.apm.plugin.activemq/define/ActiveMQProducerInstrumentation.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.plugin.activemq.define;
+
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.apache.activemq.ActiveMQSession;
```
Comment (wu-sheng) [inline apm-sniffer/apm-sdk-plugin/activemq-5.x-plugin/src/main/java/org.apache.skywalking.apm.plugin.activemq/define/ActiveMQProducerInstrumentation.java]: You imported a class, which is not forbidden to exist in Instrument class.

File: apm-sniffer/apm-sdk-plugin/activemq-5.x-plugin/src/main/java/org.apache.skywalking.apm.plugin.activemq/define/ActiveMQConsumerInstrumentation.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.plugin.activemq.define;
+
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.apache.activemq.ActiveMQSession;
```
Comment (wu-sheng) [inline apm-sniffer/apm-sdk-plugin/activemq-5.x-plugin/src/main/java/org.apache.skywalking.apm.plugin.activemq/define/ActiveMQConsumerInstrumentation.java]: Same import issue

**Comment by ascrutae** [comment]
Here is the [test report](https://github.com/SkywalkingTest/agent-integration-test-report/blob/master/2018/8/oatiz/testReport-master-2018-08-03-16-55.md) and [log](https://github.com/apache/incubator-skywalking/files/2256870/validate-2018-08-03-16-55.log)

**Comment by peng-yongsheng** [comment]
The newest [test report](https://github.com/SkywalkingTest/agent-integration-test-report/blob/master/2018/8/WithLin/testReport-master-2018-08-04-10-06.md) looks very good.
Just 2 version(5.13.4, 5.10.0) tested fail. @WithLin Do you want to fix them or after this PR merge?

File: apm-sniffer/apm-sdk-plugin/activemq-5.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/activemq/ActiveMQConsumerInterceptor.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+    public static final String OPERATE_NAME_PREFIX = "ActiveMQ/";
+    public static final String CONSUMER_OPERATE_NAME_SUFFIX = "/Consumer";
+    @Override
+    public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes, MethodInterceptResult result) throws Throwable {
+        ContextCarrier contextCarrier = new ContextCarrier();
+        String url = (String) objInst.getSkyWalkingDynamicField();
+        MessageDispatch messageDispatch = (MessageDispatch) allArguments[0];
+        AbstractSpan activeSpan = null;
+        if (messageDispatch.getDestination().getDestinationType() == 1 ||  messageDispatch.getDestination().getDestinationType() == 5) {
```
Comment (ascrutae) [inline apm-sniffer/apm-sdk-plugin/activemq-5.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/activemq/ActiveMQConsumerInterceptor.java]: The magic number make me confuse. Could you change those number to constant number?

**Comment by wu-sheng** [comment]
I suggest you send a new pr to change the number to Constant with meaning name. Such as destination type.

**Pull request merged**